### DISCUSSION TITLE
Too many idle connections

### DISCUSSION THREAD

**Discussion by guibhz** [dbeaver/dbeaver#12404] [Q&A]
**System information:**
- (Azure) PostgreSQL 11.6 build 1800, 64-bit 
- Clients are using different versions of DBeaver (between 7.2 and 21), with the option auto-commit mode enable in most of then.

**Describe the problem you're observing:**

Hello everyone, 
I'm facing a strange situation using Dbeaver with Postgresql database.

The error **"FATAL: remaining connection slots are reserved for non-replication superuser connections"** started to popup more frequently in my server.

After inspecting the Session Manager, I noticed that a lot connections state (from many users) are Idle, running the query "SHOW search_path". 

I've talked with one of the users, and he told me that he has only 2 opened scritps, but the session manager showed that he has 12 open connections, 10 of them were Idle, running this query "SHOW search_path". (see below)

![image](https://user-images.githubusercontent.com/890629/117740505-fa933080-b1d6-11eb-8512-cdb923769687.png)

**Querstion**
1. Does anyone know if this is a normal behavior, and why so many connections are being opened with this query "SHOW search_path"?
2. Is there a way to optimize the number of opened connections, and avoid (or minimize) multiple idle connections?

Thanks in advanced.

**Comment by guibhz** [comment]
Does anyone has a suggestion / solution?

**Comment by AndreasK79** [comment]
We have the same problem here. I've seen that behavior, but I have not found out why it happens. Would be nice to know why this happens, and if it can be prevented.

**Reply by guibhz** [reply]
I've tried a lot of different configurations, but none of then worked, so we created an automation to kill idle connections with pg.application_name like 'DBeaver%'. It's working fine!

**Comment by cardalda** [comment]
Anyone a solution?
We also kill the connections with a script, but it's annoying to see that sometime dbeaver decides to invalidate all connections and is reclaiming all the connections again. I have only one tab open, and can't figure out why it needs 17 connections (!)

**Reply by LonwoLonwo** [reply]
Maybe this answer will be helpful for you https://github.com/dbeaver/dbeaver/issues/8303#issuecomment-1316882186
